Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator.swift

```diff
@@ -0,0 +1,152 @@
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import JavaTypes
+
+struct JNISwift2JavaGenerator: Swift2JavaGenerator {
+  let analysis: AnalysisResult
+  let swiftModuleName: String
+  let javaPackage: String
+  let logger: Logger
+  let swiftOutputDirectory: String
+  let javaOutputDirectory: String
+
+  var javaPackagePath: String {
+    javaPackage.replacingOccurrences(of: ".", with: "/")
+  }
+
+  init(
+    translator: Swift2JavaTranslator,
+    javaPackage: String,
+    swiftOutputDirectory: String,
+    javaOutputDirectory: String
+  ) {
+    self.logger = Logger(label: "jni-generator", logLevel: translator.log.logLevel)
+    self.analysis = translator.result
+    self.swiftModuleName = translator.swiftModuleName
+    self.javaPackage = javaPackage
+    self.swiftOutputDirectory = swiftOutputDirectory
+    self.javaOutputDirectory = javaOutputDirectory
+  }
+
+  func generate() throws {
+    try writeExportedJavaSources()
+  }
+}
+
+extension JNISwift2JavaGenerator {
+  func writeExportedJavaSources() throws {
+    var printer = CodePrinter()
+
+    let filename = "\(self.swiftModuleName).java"
+    logger.trace("Printing module class: \(filename)")
+    printModule(&printer)
+
+    if let outputFile = try printer.writeContents(
+      outputDirectory: javaOutputDirectory,
+      javaPackagePath: javaPackagePath,
+      filename: filename
+    ) {
+      logger.info("[swift-java] Generated: \(self.swiftModuleName).java (at \(outputFile))")
+    }
+  }
+}
+
+extension JNISwift2JavaGenerator {
+  private func printModule(_ printer: inout CodePrinter) {
+    printHeader(&printer)
+    printPackage(&printer)
+
+    printModuleClass(&printer) { printer in
+      for decl in analysis.importedGlobalFuncs {
+        self.logger.trace("Print global function: \(decl)")
+        printFunctionBinding(&printer, decl)
+      }
+    }
+  }
+
+  private func printHeader(_ printer: inout CodePrinter) {
+    printer.print(
+      """
+      // Generated by jextract-swift
+      // Swift module: \(swiftModuleName)
+
+      """
+    )
+  }
+
+  private func printPackage(_ printer: inout CodePrinter) {
+    printer.print(
+      """
+      package \(javaPackage);
+
+      """
+    )
+  }
+
+  private func printModuleClass(_ printer: inout CodePrinter, body: (inout CodePrinter) -> Void) {
+    printer.printBraceBlock("public final class \(swiftModuleName)") { printer in
+      body(&printer)
+    }
+  }
+
+  private func printFunctionBinding(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
+    let returnType = decl.functionSignature.result.type.javaType
+    let params = decl.functionSignature.parameters.map {
+      "\($0.type.javaType) \($0.parameterName!)"
```

Comment:
Should probably safeguard the same way FFM does, but just naming them `_0`, `_1` etc.. however, I am not sure when `parameterName` actually can be `nil`? 🤔

Reply:
`func foo(_: Int)` case _should_ be `nil`. Not so common, but for example:
```swift
protocol SomeProtocol {
  init()
}
func foo<T: SomeProtocol>(_: T.Type) -> T {
  return T.init()
}
```

Reply:
yeah we should handle the nil, it's not very common but we should handle it.

We can maybe centralize it somehow to provide some one place where we do `effectiveJavaParameters()`? The impl of naming those could be just:

```
[...].enumerated().map { idx, param in
  "\(param.type.javaType) \(param.paremeterName ?? "arg\(idx)")"
}
```

At the same time maybe it's a good time to then fix them to become maybe `arg0` which is a common Java way to spell those "when names are unknown" 🤔